docs/content/concepts/sui-move-concepts/packages/automated-address-management.mdx

```diff
@@ -0,0 +1,85 @@
+
+## Package Upgrade Guidance
+
+- When [upgrading](./upgrade#upgrade-requirements), you will need to retrieve the `UpgradeCap` ID of your published package. Automated address management does not track your `UpgradeCap`.
+
+- When [upgrading](./upgrade#example), you will first need to set the `[addresses]` value for your package to `0x0` in the `Move.toml`, and restore its ID with the `ORIGINAL-ADDRESS` after upgrading.
+
+## Troubleshooting
+
+Conflicting published package addresses may happen when the state of package data is inconsistent.
+
+For example, you may have an existing package with a `published-at` value in the `Move.toml`. The package is re-published for testing purposes, and is now tracked in `Move.lock` with automated address management. The address in the `Move.toml` and `Move.lock` now differ, and there will be an error the next time you try to publish or upgrade the package.
+
+Here are steps to remediate conflicting addresses:
+
+- If the conflict is in a package you maintain:
+  - Delete the `published-at` value in your `Move.toml` if this published address is no longer needed
+  - Alternatively, follow the steps to [adopt automated address management](#adopting-automated-address-management-for-published-packages)
+
+- If the conflict is in a package that you do not maintain (e.g., a dependency), consider fixing the issue locally with the steps above, or contacting the maintainer to upstream changes.
+
+## Internal reference
+
+This section is an overview of the schema and internal operation of tracked address in the `Move.lock` file. The ordinary developer does not need to understand these internals. It is a reference for those who want a complete interface or control to internal state tracking.
+
+A concrete example of the schema and state you might find in a `Move.lock` file:
+
+```toml
+[env]
+
+[env.testnet]
+chain-id = "4c78adac"
+original-published-id = "0xa6041ac57f9151d49d00dcdc4f79f8c5ba1e399e1005dcb0fdd1c8632020d5a6"
+latest-published-id = "0xa6041ac57f9151d49d00dcdc4f79f8c5ba1e399e1005dcb0fdd1c8632020d5a6"
+published-version = "1"
+
+[env.mainnet]
+chain-id = "35834a8a"
+original-published-id = "0xaee5759aedf6c533634cdd2de412f6e6dfc754a89b0ec554a73597348f334477"
+latest-published-id = "0xaee5759aedf6c533634cdd2de412f6e6dfc754a89b0ec554a73597348f334477"
+published-version = "1"
+```
+
+Notes:
+
+- An `[env]` table contains entries for each environment. When a package is published for an active environment, an entry is added or updated.
+ 
+- An entry is added or updated based on its the `chain-id` of the environment. I.e., addresses are keyed by `chain-id`, not the `[env.NAME]` part. This is so that packages and their dependencies are resolved canonically by chain identifier: keying by `[env.NAME]` is not robust when users may choose arbitrary environment `NAME` aliases.
```

Comment:
```suggestion
- An entry is added or updated based on its the `chain-id` of the environment. As in, addresses are keyed by `chain-id`, not the `[env.NAME]` part. This is so that packages and their dependencies are resolved canonically by chain identifier: keying by `[env.NAME]` is not robust when users can choose arbitrary environment `NAME` aliases.
```